### DISCUSSION TITLE
Candle stick chart

### DISCUSSION THREAD

**Discussion by sanisloandras** [patrykandpatrick/vico#427] [Ideas]
Hello!

I really like this library, but wondering if I could customize it to have a candle stick style?
Currently trying to understand how the lib works, but it you have some time, it would help a lot...
Thanks in advance.

![image](https://user-images.githubusercontent.com/29690177/206598534-a248614e-0d8d-402b-a5c9-c1c018f51a54.png)

**Comment by patrickmichalik** [comment]
Hello! Vico doesn’t yet include such a chart type, but we plan on adding more chart types, including candlestick charts, in the future. The API is quite flexible, so I believe it would be possible to write a `Chart` implementation that draws candlesticks, but you’d have to create a custom `ChartEntry` implementation too (since here you’d need four _y_ values per entry, not just one).

**Comment by jibe-n** [comment]
Hello!

I was wondering if you have an ETA for the candlestick chart implementation? I played a bit with the candlestick chart in the sample and it looks so great already! I switched all my charts to Vico and this is the only one thing I'm missing.

Thank you guys for your amazing work.

**Comment by Gowsky** [comment]
Hi @jibe-n, the candlestick chart in the core module is mostly done. We need to add some Jetpack Compose and Views specific support for it, but our priority currently has been shifted to other features and improvements.

**Comment by makaronis** [comment]
is there any update on this? i'm looking for candlestick chart too

**Comment by Gowsky** [comment]
Hi @makaronis, no updates since my last comment. I want to resume work on candlestick chart once #87 is done, and it’s nearly complete.

**Comment by makaronis** [comment]
Hi @Gowsky , what's missing from current realization of that chart? i've seen branch with that code

**Comment by Gowsky** [comment]
The API probably needs some tweaks, and we need to add missing documentation.

**Comment by makaronis** [comment]
Hello again, i've seen that branch with candlestick was updated a long time ago, and currently out of sync with main branch. Do you have an estimation about when you have time to update that branch? I'm struggling to make merge of this branch by my own, and i can't use current version because it missing itemPlacer api

**Reply by patrickmichalik** [reply]
Sorry for the late reply. It looks like GitHub unsubscribed me from this thread when I converted it from an issue into a discussion. Work on candlestick charts and pie charts has been resumed—`master` and `candlestick-and-pie-charts` are generally in sync.

**Reply by JanimeSenpai** [reply]
Hi!
Chart 10 in the example app is now a Candlestick chart, but I couldn't import rememberCandlestickCartesianLayer in my own project. I am working with version 2.0.0-alpha.14.
I got the error "Unresolved reference: rememberCandlestickCartesianLayer"

**Reply by Gowsky** [reply]
Hello, candlestick charts are not released yet. Stay tuned!

**Reply by patrickmichalik** [reply]
Hello, @JanimeSenpai! I wanted to expand on @Gowsky’s comment above. Candlestick charts are planned for Vico 2.0.0 Alpha 15, which we’re finalizing on the `master` branch. When accessing the sample app, please ensure that you have the repository checked out at the right tag. For Vico 2.0.0 Alpha 14, the tag is `v2.0.0-alpha.14`.

**Comment by patrickmichalik** [comment]
[Vico 2.0.0 Alpha 15](https://github.com/patrykandpatrick/vico/releases/tag/v2.0.0-alpha.15) introduced candlestick charts. Cheers!

**Comment by Gowsky** [comment]
Hello. [Vico 2.0.0](https://github.com/patrykandpatrick/vico/releases/tag/v2.0.0) is now available.
